Tests/SystemTests/NetworkConnectivityTests/Subscriber/SubscriberNetworkConnectivityTests.swift

```diff
@@ -0,0 +1,384 @@
+                                case .failure(let error):
+                                    handler(.failure(error))
+                                }
+                            }
+                        case .failure(let error):
+                            handler(.failure(error))
+                        }
+                    }
+                case .failure(let error):
+                    handler(.failure(error))
+                }
+            }
+        }
+
+        // Resolve the fault, wait for Trackable to move to expected state
+
+        let thirdLocationUpdate = Location(coordinate: .init(latitude: 4, longitude: 4))
+        let thirdPublisherResolution = Resolution(accuracy: .maximum, desiredInterval: 3, minimumDisplacement: 0)
+
+        try subscriberMonitorFactory.forResolvedFault(
+            label: "[fault resolved] subscriber",
+            locationUpdate: thirdLocationUpdate,
+            publisherResolution: thirdPublisherResolution,
+            subscriberResolution: secondSubscriberResolution
+        ).waitForStateTransition { (logHandler, handler: @escaping (Result<Void, Error>) -> Void) in
+            defaultAbly.updatePresenceData(
+                trackableId: testResources.trackableID,
+                presenceData: .init(type: .publisher, resolution: thirdPublisherResolution)
+            ) { [testResources] result in
+                switch result {
+                case .success:
+                    defaultAbly.sendEnhancedLocation(
+                        locationUpdate: .init(location: thirdLocationUpdate),
+                        trackable: .init(id: testResources.trackableID)
+                    ) { result in
+                        switch result {
+                        case .success:
+                            logHandler.debug(message: "Sent third enhanced location update on Ably channel", error: nil)
+
+                            // Resolve the problem
+                            testResources.resolveFaultAsync(handler)
+                        case .failure(let error):
+                            handler(.failure(error))
+                        }
+                    }
+                case .failure(let error):
+                    handler(.failure(error))
+                }
+
+            }
+        }
+
+        // Restart the fault to simulate the publisher going away whilst we're offline
+
+        try subscriberMonitorFactory.forActiveFault(
+            label: "[fault active] publisher shutdown for disconnect test",
+            locationUpdate: thirdLocationUpdate,
+            publisherResolution: /* thirdPublisherResolution, */ nil, /* TODO I’m not convinced that thirdPublisherResolution is right. Since shutdownAblyPublishing leaves presence with a hardcoded presence data, and hence a hardcoded resolution (accuracy: .balanced, desiredInterval: 1, minimumDisplacement: 0) that’s the only resolution the subscriber can expect to definitely receive */
```

Comment:
@AndyTWF I was wondering if you had any thoughts on this — did I miss something?

Reply:
I believe in Android it was because the last thing on the StateFlow was the third resolution, so checking it again would result in the same value being emitted to a new collector?

Reply:
But `testResources.shutdownAblyPublishingAsync` causes the publisher to emit a resolution update, doesn't it? Wouldn't we expect that to be received by the subscriber?

Reply:
So yes, but it never reaches the subscribers stateflow.

As the presence event is a `LEAVE`, the subscriber doesn't actually process it as a resolution update, it just marks the publisher as gone (see `updateForPresenceMessagesAndThenEmitStateEventsIfRequired` in Android).

So I would expect the "last" emitted value to be that of the last presence `update` - which in this case is the `thirdPublisherResolution`.

Reply:
Cool, thanks for the explanation. I've opened https://github.com/ably/ably-asset-tracking-swift/pull/633 which ignores `ABSENT` and `LEAVE` actions containing resolutions, and that makes the original assertion work.